### PR TITLE
refactor(crd-generator): deprecation of CRD Generator v1

### LINKED ISSUE #6673: CRDGenertor: Deprecate v1
## Description

We have provided a feature-par replacement for CRD Generator v1.
In version 7.0.0, we should mark `v1` as deprecated (for removal) and suggest users to migrate to the new one.

Since the CRD Generator v1 works as an annotation processor we should add a warning message after the CRD has been generated pointing users to migrate to the new generator.

### PR DESCRIPTION
**Pull request by manusa**
## Description

Fixes #6673

/cc @baloo42 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by metacosm — APPROVED** [review]

File: doc/CRD-generator.md [hunk 1]
```diff
@@ -1,27 +1,42 @@
 # CRD generation
 
+The CRD Generator annotation processing tool (APT) (`io.fabric8:crd-generator-apt`) and its API (`io.fabric8:crd-generator-api`) are being deprecated and will eventually be removed once we offer a complete replacement for all users.
+
+As a replacement, we're currently providing a new version of the API in the `io.fabric8:crd-generator-api-v2`and a few tools to be able to leverage it in your projects.
+
+The following list contains the available tooling:
+- CRD Generator Maven Plugin: A Maven plugin that generates CRDs during the build process.
```
Comment (baloo42) [inline doc/CRD-generator.md:8]:
```suggestion
- [CRD Generator Maven Plugin](../crd-generator/maven-plugin/README.md): A Maven plugin that generates CRDs during the build process.
```

File: doc/CRD-generator.md [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
+- CRD Generator CLI tool: A CLI tool that generates CRDs when executed.
```
Comment (baloo42) [inline doc/CRD-generator.md:9]:
```suggestion
- [CRD Generator CLI tool](../crd-generator/cli/README.md): A CLI tool that generates CRDs when executed.
```

File: doc/CRD-generator.md [hunk 3]
```diff
@@ -1,27 +1,42 @@
...
+
 ## Quick start
 
-Import the Annotation Processor into your build.
+Add the CRD Generator plugin to your project or use the CLI tool:
```
Comment (baloo42) [inline doc/CRD-generator.md:13]:
```suggestion
Add the CRD Generator plugin to your project:
```

Comment (baloo42) [reply doc/CRD-generator.md:13]: In my opinion in the quickstart we should focus on the maven and gradle plugin. Compared to those tools the CLI will be less common and you already provided a complete list of the tooling in the beginning.

File: doc/CRD-generator.md (hunk 1 above)
Comment (manusa) [reply doc/CRD-generator.md]: Nice, I missed those, thx!

File: doc/CRD-generator.md (hunk 2 above)
Comment (manusa) [reply doc/CRD-generator.md]: Thx!

**Comment by Tanemahuta** [comment]
Hello there!

I have a working implementation ready, I need to add tests, and ask for permission to contribute from my employer.
Interested?

**Comment by shawkins** [comment]
@Tanemahuta do you mean contribute another CRD generator implemenation? This is a related, but closed pr - you should start a new discussion for that, or look to see how your changes could be incorporated into the v2 version.

**Comment by Tanemahuta** [comment]
Sorry, wrong issue, too many windows open. Will comment on the correct one.

**Pull request merged**